central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -0,0 +1,244 @@
+package complianceReportgenerator
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/pkg/errors"
+	checkResults "github.com/stackrox/rox/central/complianceoperator/v2/checkresults/datastore"
+	"github.com/stackrox/rox/central/graphql/resolvers"
+	"github.com/stackrox/rox/central/graphql/resolvers/loaders"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/branding"
+	"github.com/stackrox/rox/pkg/csv"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/grpc/authz/allow"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/notifier"
+	"github.com/stackrox/rox/pkg/notifiers"
+	"github.com/stackrox/rox/pkg/retry"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/templates"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	reportGenCtx = resolvers.SetAuthorizerOverride(loaders.WithLoaderContext(sac.WithAllAccess(context.Background())), allow.Anonymous())
+
+	csvHeader = []string{
+		"Control Reference",
+		"Check(CCR)",
+		"Profile(version)",
+		"Check Description",
+		"Cluster",
+		"Status",
+		"Remediation",
+	}
+)
+
+const (
+	maxNumberProfilesinSubject = 4
+)
+
+type formatBody struct {
+	BrandedPrefix    string
+	Profile          string
+	Pass             int
+	Fail             int
+	Mixed            int
+	Cluster          int
+	ComplianceStatus string
+}
+
+type formatSubject struct {
+	BrandedPrefix string
+	ScanConfig    string
+	Profiles      string
+}
+
+type complianceReportGeneratorImpl struct {
+	checkResultsDS        checkResults.DataStore
+	notificationProcessor notifier.Processor
+}
+
+// struct which hold all columns of a row
+type resultRow struct {
+	ClusterName string
+	CheckName   string
+	Profile     string
+	ControlRef  string
+	Description string
+	Status      string
+	Remediation string
+}
+
+type resultEmail struct {
+	resultCSVs map[string][]*resultRow //map of cluster id to slice of *resultRow
```

Comment:
This can get very large, need to vet the scale issues

Reply:
@dashrews78 ^

Reply:
yeah.  I'm not sure what to do about this yet.  @clickboo Would it make sense to for @ksurabhi91 to write a ticket to investigate so that we can move forward?  I would assume that any such export/report would have the same issue.  So hopefully we have solved this before.